api/python/quilt3/packages.py

```diff
@@ -340,6 +341,26 @@ def __init__(self, registry, name, top_hash):
         self.top_hash = top_hash
 
 
+class ManifestJSONDecoder(json.JSONDecoder):
+    """
+    Standard json.JSONDecoder reuses same `str` objects for JSON properties, while doing
+    a single `decode()` call.
+    This class also reuses `str` between many `decode()`s.
+    """
+    def __init__(self, *args, **kwargs):
+        @functools.lru_cache(maxsize=None)
```

Comment:
"If maxsize is set to None, the LRU feature is disabled and the cache can grow without bound."—don't we want to bound this ourselves?

Reply:
Can you also add a comment as to how exactly this is saving memory? I must not be understanding something, because reading of object_pairs_hook is that this will only save us memory when the same raw object is decoded twice, but that should rarely happen?

Reply:
```python
In [48]: def test(decoder):
    ...:     data = (
    ...:         '{"key": "value"}',
    ...:         '{"key": "value"}',
    ...:         '{"key": "other value"}',
    ...:     )
    ...:     str_objects = {}
    ...:     for obj in jsonlines.Reader(data, loads=decoder.decode):
    ...:         for item in obj.items():
    ...:             for s in item:
    ...:                 str_objects[id(s)] = s
    ...:     return str_objects
    ...: 

In [49]: test(json.JSONDecoder())
Out[49]: 
{140254112745488: 'key',
 140254112744760: 'value',
 140254114859584: 'key',
 140254114858912: 'value',
 140254114860256: 'key',
 140254113566256: 'other value'}

In [50]: test(ManifestJSONDecoder())
Out[50]: 
{140254113989160: 'key',
 140254113988936: 'value',
 140254113991960: 'value',
 140254105343920: 'other value'}
```
We always have `type` and `value` keys for hash objects that end up in `Package` instance. Also the package from the PR description uses uniform schema for meta of all objects, i.e. there are a lot of keys that could be reused.